### ISSUE TITLE
Mixture of Tokens uses Dirichlet/subset actions and a proxy likelihood, not Gumbel reparameterization

### ISSUE THREAD

**Issue by microprediction** [microprediction/latentreasoning#73]
## Reviewed source

Commit: [`11b700ba0c9cc03b0e1a54db15fe06d15fad8e51`](https://github.com/microprediction/latentreasoning/commit/11b700ba0c9cc03b0e1a54db15fe06d15fad8e51)

Affected text:

- [`training.html` lines 259–263](https://github.com/microprediction/latentreasoning/blob/11b700ba0c9cc03b0e1a54db15fe06d15fad8e51/training.html#L259-L263) puts both SofT-GRPO and Jain–Rappazzo’s Mixture of Tokens in the “Logits” row, describes Gumbel noise and the Gumbel density of recovered noise, and labels the estimator “GRPO with Gumbel reparameterization.”
- [`training.html` lines 346–349](https://github.com/microprediction/latentreasoning/blob/11b700ba0c9cc03b0e1a54db15fe06d15fad8e51/training.html#L346-L349) more accurately calls Mixture of Tokens a Dirichlet-weighted method.
- [`bibliography.html` lines 1134–1136](https://github.com/microprediction/latentreasoning/blob/11b700ba0c9cc03b0e1a54db15fe06d15fad8e51/bibliography.html#L1134-L1136) likewise correctly identifies Dirichlet-weighted or multi-sampled mixtures, but does not mention the approximate likelihood.

## Mixture of Tokens has no Gumbel action

The current [Jain and Rappazzo paper, arXiv:2509.21482v1](https://arxiv.org/html/2509.21482) defines two experimental MoT-G variants in §3.2:

1. **Dirichlet:** take the top-`k` tokens deterministically, then sample their mixture weights from a Dirichlet distribution parameterized by normalized token probabilities.
2. **Different Tokens:** sample `k` tokens without replacement, then weight their embeddings by normalized token probabilities.

There is no Gumbel perturbation in the paper. Searching the full current text for “Gumbel” returns no occurrence. The stochastic action is either a Dirichlet weight vector or a sampled token subset—not Gumbel noise added to logits.

## The paper does not use an exact action-density ratio

Section 3.1 explicitly says that loss computation is subtle. For a mixture trajectory, the desired probability of the sampled token set can be combinatorially expensive, especially for sampling without replacement. It offers:

- a **single-token loss**, using the log probability of one sampled component token;
- a **multiple-token loss**, propagating to all sampled component tokens; and
- for stabilized/off-policy GRPO, a plug-in approximation of the form

```text
log pi_theta(trajectory)
  approximately equals
sum_{m,n} p_n(z_{m,n}) log p_n(z_{m,n})
```

That is not the log density of the Dirichlet weight action, nor the exact probability of an unordered without-replacement token set. The method therefore uses component-token surrogate/plug-in scores, not a recovered-noise Gumbel density and not a Gumbel reparameterization estimator.

This distinction matters because the page introduces the table by saying noise location “fixes the form of the log-probability.” For MoT-G, the paper itself says the true mixture-trajectory likelihood is difficult and substitutes an approximation.

## Consequence

The table conflates two materially different algorithms:

- SofT-GRPO perturbs logits with Gumbel noise and scores that perturbation (with the separate estimator qualification already raised in [#66](https://github.com/microprediction/latentreasoning/issues/66)).
- Mixture of Tokens samples Dirichlet weights or token subsets and trains with component-token surrogate/approximate likelihoods.

As written, readers are told that MoT-G has a tractable Gumbel action density and corresponding GRPO ratio that the paper neither defines nor uses.

## Suggested correction

Give Mixture of Tokens a separate row:

| Noise/action space | Action | Estimator |
|---|---|---|
| Mixture weights or token subset | Top-`k` embeddings with sampled Dirichlet weights; or `k` tokens sampled without replacement and probability-weighted | GRPO using single-/multiple-component token losses and a plug-in trajectory-likelihood approximation; not an exact Gumbel density |

The prose and bibliography can retain the reported Reasoning-Gym results and hidden-state-entropy observation, while adding that its policy likelihood is approximate rather than treating it as the same estimator as SofT-GRPO.

**Comment by microprediction** [comment]
Correct, fixed in 1185867.

The word Gumbel does not occur anywhere in arXiv:2509.21482. The paper defines two variants, Dirichlet-sampled weights over greedy top-k tokens, and k tokens sampled without replacement weighted by normalised probability. For the likelihood it says the exact trajectory probability is "combinatorially expensive to compute (especially if sampling without replacement)" and uses a plug-in approximation instead.

It had been sharing a table row with SofT-GRPO that described a mechanism it does not have. It now sits with the discrete-sampling methods with its own description, and the prose elsewhere on the page, which already said Dirichlet, no longer contradicts the table.
